### ISSUE TITLE
flutter build on ios fails

### ISSUE THREAD

**Issue by daadu** [freshworks/freshchat-flutter-sdk#1]
Throwing following error
```
[!] The 'Pods-Runner' target has transitive dependencies that include statically linked binaries: (/Users/user/AndroidStudioProjects/my_app/ios/Pods/FreshchatSDK/FreshchatSDK/libFDFreshchatSDK.a)
```

**Comment by daadu** [comment]
Any workaround that can be suggested? Until fixed in next release.

**Comment by daadu** [comment]
also I think, the code should be made public.

**Comment by daadu** [comment]
```podspec
target 'Runner' do
  #use_frameworks!
  use_modular_headers!

  flutter_install_all_ios_pods File.dirname(File.realpath(__FILE__))
end
```

commenting out `use_frameworks!` did the job, but i think the dependcies of pod needs to be properly configured (i am new to iOS, I could be wrong.)

**Comment by daadu** [comment]
but the above workaround is breaking other plugins.

**Comment by daadu** [comment]
- added `s.static_framework = true` in "cached" plugin `freshchat_sdk.podspec` file
```podspec
Pod::Spec.new do |s|
  ...
  s.static_framework = true
end
```
- this resolved the "statically linked" isssue but then "Include of non-modular header inside framework module" error, i "tried" to  fixed it by "allowing non-moduler header"
<img width="914" alt="Screenshot 2021-03-13 at 1 22 12 PM" src="https://user-images.githubusercontent.com/4963236/111023536-a4d12080-83ff-11eb-80bb-0d4e83c90279.png">
- now, got the following error in build

   ```
    1 warning and 1 error generated.
    /Users/harsh/AndroidStudioProjects/next_home/ios/Runner/GeneratedPluginRegistrant.m:68:9: fatal error: could not build module 'freshchat_sdk'
    #import <freshchat_sdk/FreshchatSdkPlugin.h>
     ~~~~~~~^
    1 warning and 2 errors generated.
    note: Using new build system
    note: Building targets in parallel
    note: Planning build
    note: Constructing build description
    ```
- at this point I am not sure my "fixes/workaround" were proper.

**Comment by daadu** [comment]
- reverted back to not ""allowing non-moduler includes"
<img width="883" alt="Screenshot 2021-03-13 at 1 37 54 PM" src="https://user-images.githubusercontent.com/4963236/111023848-53c22c00-8401-11eb-9ae7-c40b36fc58d2.png">

- moved the following  "includes" from `FreshchatSdkPlugin.h` to `FreshchatSdkPlugin.m`:
```
#if __has_include("FreshchatSDK.h")
#import "FreshchatSDK.h"
#else
#import "FreshchatSDK/FreshchatSDK.h"
#endif
```

- that worked and build was successfull

**Comment by daadu** [comment]
- when "identiyingUser" and "startConversation" (not sure where exactly) the app crashed with following logs
<img width="817" alt="Screenshot 2021-03-13 at 1 41 49 PM" src="https://user-images.githubusercontent.com/4963236/111023947-04c8c680-8402-11eb-9694-657baae0ee66.png">

**Comment by daadu** [comment]
The above error was resolved, by passing null with `FreshchatUser(null, null)`.

Now the app crashes, after `startConversation` without any error in logs.

**Comment by daadu** [comment]
The issue was not calling `setLastName` with `FreshchatUser`, passing `""` now, resolved this issue.

**Comment by carman247** [comment]
Add this to my Podfile solved the issue for me

```
pre_install do |installer|
  # workaround for https://github.com/CocoaPods/CocoaPods/issues/3289
  Pod::Installer::Xcode::TargetValidator.send(:define_method, :verify_no_static_framework_transitive_dependencies) {}
end
```

**Comment by matszafraniec** [comment]
Thank you @daadu . I have same issue. 

Can we expect any respond from Freshchat team?

**Comment by daadu** [comment]
@matszafraniec I fixed it in #3. You can use that fork version in you branch, until merged.

**Comment by rohit-ramkumar** [comment]
Hi @daadu @matszafraniec 
Thank you for pointing this out, 
These issues have been fixed as part of the latest version (0.5.0)
Closing this issue for now.
